src/gwt/org/bbop/apollo/gwt/client/AnnotatorPanel.java

```diff
@@ -688,9 +688,28 @@ public void displayTranscript(int geneIndex, String uniqueName) {
         MainPanel.updateGenomicViewerForLocation(selectedAnnotationInfo.getSequence(), min, max, false);
     }
 
+    // also used by javascript function
+    public void displayFeature(int featureIndex, String uniqueName) {
+        AnnotationInfo annotationInfo = dataGrid.getVisibleItem(Math.abs(dataGrid.getVisibleRange().getStart() - featureIndex));
+        String type = annotationInfo.getType();
+        if (type.equals("transposable_element") || type.equals("repeat_region")) {
+            // do nothing
+        }
+        else {
+            exonDetailPanel.updateData(annotationInfo);
+        }
+        gotoAnnotation.setEnabled(true);
```

Comment:
Even though this is a bit duplicated, for some reason it works better that way when called directly from JavaScript.